File: news-api/src/main/java/com/newsarcticlesapp/newsapi/model/analytics/GeneralAnalytics.java
package com.newsarcticlesapp.newsapi.model.analytics;

import lombok.Data;

import java.util.List;

@Data
public class GeneralAnalytics {
    private Long amountOfArticles;
    private List<TopicsCount> topicsCounts;
    private List<KeywordCount> keywordCounts;
    private List<SourceCount> sourceCounts;
}

File: application-ui/src/app/models/analytics/general-analytics.ts
import {TopicsCount} from './topics-count';
import {KeywordCount} from './keyword-count';
import {SourceCount} from './source-count';

export class GeneralAnalytics {
  amountOfArticles: number;
  topicsCounts: TopicsCount[];
  keywordCounts: KeywordCount[];
  sourceCounts: SourceCount[];
}
